runpod/serverless/modules/worker_state.py

```diff
@@ -40,30 +22,36 @@ def get_auth_header():
     return {"Authorization": f"{os.environ.get('RUNPOD_AI_API_KEY')}"}
 
 
-def get_current_job_id():
-    '''
-    Returns the current job id.
-    '''
-    return CURRENT_JOB_ID
 
+class Jobs:
+    ''' Track the state of current jobs.'''
 
-def get_done_url():
-    '''
-    Constructs the done URL using the current job id.
-    '''
-    return JOB_DONE_URL_TEMPLATE.replace('$ID', CURRENT_JOB_ID)
+    _instance = None
+    jobs = []
 
+    def __new__(cls):
+        if Jobs._instance is None:
+            Jobs._instance = object.__new__(cls)
+            Jobs._instance.jobs = []
+        return Jobs._instance
 
-def get_stream_url():
-    '''
-    Constructs the stream URL using the current job id.
-    '''
-    return JOB_STREAM_URL_TEMPLATE.replace('$ID', CURRENT_JOB_ID)
+    def add_job(self, job_id):
+        '''
+        Adds a job to the list of jobs.
+        '''
+        self.jobs.append(job_id)
```

Comment:
@Jorghi12 What if we do `self.jobs = set(self.jobs.append(job_id))` here to enforce id uniqueness?

Reply:
Additionally, we can check if a job is already in the list when it does get_job

Reply:
Interesting. Sure, we could initialize `self.jobs = set()` and do `self.jobs.add(job_id)` and then yeah checking if the job is already in the list when doing `get_job` works!